src/ui/setup/index.ts

```diff
@@ -76,5 +76,14 @@ export async function bootstrapApp() {
     initLaunchDarkly();
   });
 
+  // Only initialize the layout state once the apollo client is ready.
+  clientWaiter.promise.then(async () => {
+    const layoutState = await getInitialLayoutState();
+    store.dispatch(setLayout(layoutState));
+    // We don't start setting things into local preferences until we've
+    // finished fully initializing the state.
+    registerStoreObserver(store, updatePrefs);
+  });
```

Comment:
this feels like a smell on many levels. 

why can't we set the state before apollo is setup?

Is there a way to kick off an action from a react component once apollo is setup?

Reply:
cc @jcmorrow

Reply:
Yeah

The [state's still set up before apollo is set up](https://github.com/RecordReplay/devtools/blob/master/pages/_app.tsx#L137-L157). If you're asking about `registerStoreObserver`, it's pushed to here because otherwise it would create a local session immediately. If it does, then by the time we run getInitialLayoutState, there's already a local session for that replay, even if it's the user's first time opening it.

This code is late night spaghetti though so @jcmorrow if you've got ideas let me know!